File: app/src/main/java/com/xmartlabs/template/common/GsonExclude.java
package com.xmartlabs.template.common;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * Annotation to exclude a field from being serialized
 * Provides a configuration to choose whether to exclude the field of being serialized on:
 * <ul>
 *   <li>Service related operations</li>
 *   <li>Database related operations</li>
 *   <li>Both</li>
 * </ul>
 */
@Retention(RUNTIME)
@Target(FIELD)
public @interface GsonExclude {
  Strategy strategy() default Strategy.ALL;

  enum Strategy {
    ALL,
    DATA_BASE,
    SERVICE,
  }
}

File: core/src/main/java/com/xmartlabs/bigbang/core/common/GsonExclude.kt
package com.xmartlabs.bigbang.core.common


/**
 * Annotation to exclude a field from being serialized.
 * Provides a configuration to choose whether to exclude the field of being serialized on:
 *
 *  * Service related operations
 *  * Database related operations
 *  * Both
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FIELD)
annotation class GsonExclude(val strategy: GsonExcludeStrategy = GsonExcludeStrategy.ALL)

enum class GsonExcludeStrategy {
  ALL,
  DATA_BASE,
  SERVICE,
}
